sequencer/src/chain_config.rs

```diff
@@ -50,9 +70,36 @@ impl ChainConfig {
             base_fee: base_fee.into(),
         }
     }
+
+    pub fn from_toml(toml: &toml::Value) -> anyhow::Result<Self> {
```

Comment:
Couldn't we use toml with serde as described here: https://docs.rs/toml/latest/toml/#deserialization-and-serialization to reduce the amount of code we need?

Reply:
We could. I wanted to allow both integers and strings with units for sizes, and ended up writing some custom code for that, and also for the `U256` stuff because the default parsing for `U256` only accepts hex strings (event without a `0x` prefix and only decimal digits, it's interpreted as hex!).

But thinking about it since we already have custom wrappers for everything (`FeeAmount`, `FeeAccount`, `ChainId`) I could probably just override the deserialization for those types specifically. I can try this when I get some time.

Reply:
Done in [b193fd5](https://github.com/EspressoSystems/espresso-sequencer/pull/1443/commits/b193fd530cb0504ab7430c935f4f1a3ed2388362)